lib/OptimizationManopt/test/runtests.jl

```diff
@@ -0,0 +1,140 @@
+    opt = OptimizationManopt.GradientDescentOptimizer(R2,
+        stepsize = stepsize)
+
+    optprob_forwarddiff = OptimizationFunction(rosenbrock, Optimization.AutoEnzyme())
+    prob_forwarddiff = OptimizationProblem(optprob_forwarddiff, x0, p; manifold = R2)
+    sol = Optimization.solve(prob_forwarddiff, opt)
+    @test sol.minimum < 0.2
+
+    optprob_grad = OptimizationFunction(rosenbrock; grad = rosenbrock_grad!)
+    prob_grad = OptimizationProblem(optprob_grad, x0, p; manifold = R2)
+    sol = Optimization.solve(prob_grad, opt)
+    @test sol.minimum < 0.2
+end
+
+@testset "Nelder-Mead" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.NelderMeadOptimizer(R2)
+
+    optprob = OptimizationFunction(rosenbrock)
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt)
+    @test sol.minimum < 1e-6
+end
+
+@testset "Conjugate gradient descent" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    stepsize = Manopt.ArmijoLinesearch(R2)
+    opt = OptimizationManopt.ConjugateGradientDescentOptimizer(R2,
+        stepsize = stepsize)
+
+    optprob = OptimizationFunction(rosenbrock, Optimization.AutoForwardDiff())
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt)
+    @test sol.minimum < 0.5
+end
+
+@testset "Quasi Newton" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.QuasiNewtonOptimizer(R2)
+
+    optprob = OptimizationFunction(rosenbrock, Optimization.AutoForwardDiff())
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt)
+    @test sol.minimum < 1e-14
+end
+
+@testset "Particle swarm" begin
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+
+    opt = OptimizationManopt.ParticleSwarmOptimizer(R2)
+
+    optprob = OptimizationFunction(rosenbrock)
+    prob = OptimizationProblem(optprob, x0, p; manifold = R2)
+
+    sol = Optimization.solve(prob, opt)
+    @test sol.minimum < 0.1
+end
+
+@testset "Custom constraints" begin
+    cons(res, x, p) = (res .= [x[1]^2 + x[2]^2, x[1] * x[2]])
+
+    x0 = zeros(2)
+    p = [1.0, 100.0]
+    opt = OptimizationManopt.GradientDescentOptimizer(R2)
+
+    optprob_cons = OptimizationFunction(rosenbrock; grad = rosenbrock_grad!, cons = cons)
+    prob_cons = OptimizationProblem(optprob_cons, x0, p)
+    @test_throws SciMLBase.IncompatibleOptimizerError Optimization.solve(prob_cons, opt)
+end
+
+@testset "SPD Manifold" begin
+    M = SymmetricPositiveDefinite(5)
+    m = 100
+    σ = 0.005
+    q = Matrix{Float64}(I, 5, 5) .+ 2.0
+    data2 = [exp(M, q, σ * rand(M; vector_at = q)) for i in 1:m]
+
+    f(M, x, p = nothing) = sum(distance(M, x, data2[i])^2 for i in 1:m)
+    f(x, p = nothing) = sum(distance(M, x, data2[i])^2 for i in 1:m)
+
+    optf = OptimizationFunction(f, Optimization.AutoForwardDiff())
+    prob = OptimizationProblem(optf, data2[1]; manifold = M, maxiters = 1000)
```

Comment:
While `optf` might not even make sense on a lot of manifolds, I like this formulation here, that the manifold became a keyword argument. One of these tests might even serve as a goo example.